### DISCUSSION TITLE
Error when compile lammps with user-deepmd

### DISCUSSION THREAD

**Discussion by FeynmanBongo** [deepmodeling/deepmd-kit#333] [Q&A]
dear deepmd-kit user and  developer,
I have tried to install LAMMPS's DeePMD-kit module for weeks, so far I have obtained USER-DEEPMD module  according to the developer's instructions, however, when I continue to build LAMMPS with the USER-DEEPMD module I always encount the following error:
mpicxx -g -O3   -DLAMMPS_GZIP -DLAMMPS_MEMALIGN=64 -std=c++11 -DHIGH_PREC   -I/home/zhongxing/Software/TF/include -I/home/zhongxing/Software/TF/include -I/home/zhongxing/Software/DP/include/deepmd  -DMPICH_SKIP_MPICXX -DOMPI_SKIP_MPICXX=1    -c ../run.cpp
../pppm_dplr.cpp: In member function ‘virtual void LAMMPS_NS::PPPMDPLR::compute(int, int)’:
../pppm_dplr.cpp:124:3: **error: ‘gc’ was not declared in this scope**
   gc->reverse_comm_kspace(this,1,sizeof(FFT_SCALAR),REVERSE_RHO,
   ^~
../pppm_dplr.cpp:125:27: **error: ‘gc_buf1’ was not declared in this scope**
                           gc_buf1,gc_buf2,MPI_FFT_SCALAR);
                           ^~~~~~~
../pppm_dplr.cpp:125:35: **error: ‘gc_buf2’ was not declared in this scope**
                           gc_buf1,gc_buf2,MPI_FFT_SCALAR);
                                   ^~~~~~~
mpicxx -g -O3   -DLAMMPS_GZIP -DLAMMPS_MEMALIGN=64 -std=c++11 -DHIGH_PREC   -I/home/zhongxing/Software/TF/include -I/home/zhongxing/Software/TF/include -I/home/zhongxing/Software/DP/include/deepmd  -DMPICH_SKIP_MPICXX -DOMPI_SKIP_MPICXX=1    -c ../group.cpp
Makefile:104: recipe for target 'pppm_dplr.o' failed
make[1]: *** [pppm_dplr.o] Error 1
make[1]: *** Waiting for unfinished jobs....
../group.cpp: In member function ‘void LAMMPS_NS::Group::read_restart(FILE*)’:
../group.cpp:738:21: warning: ignoring return value of ‘size_t fread(void*, size_t, size_t, FILE*)’, declared with attribute warn_unused_result [-Wunused-result]
   if (me == 0) fread(&ngroup,sizeof(int),1,fp);
                ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~
../group.cpp:750:23: warning: ignoring return value of ‘size_t fread(void*, size_t, size_t, FILE*)’, declared with attribute warn_unused_result [-Wunused-result]
     if (me == 0) fread(&n,sizeof(int),1,fp);
                  ~~~~~^~~~~~~~~~~~~~~~~~~~~
../group.cpp:754:25: warning: ignoring return value of ‘size_t fread(void*, size_t, size_t, FILE*)’, declared with attribute warn_unused_result [-Wunused-result]
       if (me == 0) fread(names[i],sizeof(char),n,fp);
                    ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~
make[1]: Leaving directory '/home/zhongxing/Software/DPMD/lammps/src/Obj_mpi'
Makefile:201: recipe for target 'mpi' failed
make: *** [mpi] Error 2



AND the screen shot is:
![image](https://user-images.githubusercontent.com/56663400/105658039-a5216500-5f00-11eb-89d9-fa10b28d162d.png)

 I have referred to the previous discussion so the KSPACE package is "make-yes", but I have no idea how to deal with those errors. 
Sincerely hope someone can help.
Thanks a lot!!

**Comment by njzjz** [comment]
Which version are you using? For the latest deepmd-kit code, we only support LAMMPS stable_29Oct2020 (or later).

**Reply by FeynmanBongo** [reply]
Aha! This really helped me solve the problem. Thank you very much！
In the end I successfully built lammps with USER-DEEPMD. But the following errors occurs when I try to execute example "water" program with lammps.:
2021-01-26 20:49:09.714522: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x55d7da26b110 executing computations on platform Host. Devices:
2021-01-26 20:49:09.714570: I tensorflow/compiler/xla/service/service.cc:175]   StreamExecutor device (0): <undefined>, <undefined>
2021-01-26 20:49:09.714593: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x5598c575e110 executing computations on platform Host. Devices:
2021-01-26 20:49:09.714616: I tensorflow/compiler/xla/service/service.cc:175]   StreamExecutor device (0): <undefined>, <undefined>
2021-01-26 20:49:09.714761: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x5583abc14110 executing computations on platform Host. Devices:
2021-01-26 20:49:09.714791: I tensorflow/compiler/xla/service/service.cc:175]   StreamExecutor device (0): <undefined>, <undefined>
2021-01-26 20:49:09.716263: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x55e303092110 executing computations on platform Host. Devices:
2021-01-26 20:49:09.716338: I tensorflow/compiler/xla/service/service.cc:175]   StreamExecutor device (0): <undefined>, <undefined>
Not found: No attr named 'T' in NodeDef:
	 [[{{node DescrptSeA}}]]
	 [[DescrptSeA]]
Not found: No attr named 'T' in NodeDef:
	 [[{{node DescrptSeA}}]]
	 [[DescrptSeA]]
Not found: No attr named 'T' in NodeDef:
	 [[{{node DescrptSeA}}]]
	 [[DescrptSeA]]

**Screenshot**

![error1_画板 1](https://user-images.githubusercontent.com/56663400/105848510-0557fa00-601a-11eb-89ce-b5dad4990cdc.png)

 **Environment:**
ubuntu 18.04
deepmd-kit: 1.2.1
Tensorflow's python interface: python 3.8, tensorflow 2.3.0
tensorflow's C++ interface: bazel-0.24.1, tensorflow 1.14.0
lammps 29 Oct 2020

I notice that the inconsistent tensorflow version but cannot ensure if this is the problem. Based on Q&A #248，I take action as follows: Keep the Tensorflow's python interface as 2.3.0, and try to compile a C++ interface in 2.3.0, too. But after build tensorflow 2.3.0 with **bazel 3.1.0**, I cannot find the  **bazel-genfiles** in the directory so it is hard to move on. 
Could you give some suggestions again? Thank you for your help.

**Reply by njzjz** [reply]
#316? You are using the inconsistent deepmd-kit version for python training (v1.2) and c++ lammps (v1.3). Please use the v1.3 model, or v1.2 to build LAMMPS.

**Reply by FeynmanBongo** [reply]
It works！ I really appreciate your help.
